src/HelperTraits/SyncTrait.php

```diff
@@ -0,0 +1,135 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\HelperTraits;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\MigrateGTIN;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsAwareTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+use Exception;
+use WC_Product;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Trait with some helpers for API Pull and MC Push Sync modes.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\HelperTraits
+ */
+trait SyncTrait {
+
+	use OptionsAwareTrait;
+
+	/**
+	 * Get the products datatype key
+	 *
+	 * @return string
+	 */
+	protected function get_products_datatype(): string {
+		return 'products';
+	}
+
+	/**
+	 * Get the coupons datatype key
+	 *
+	 * @return string
+	 */
+	protected function get_coupons_datatype(): string {
+		return 'coupons';
+	}
+
+	/**
+	 * Get the shipping datatype key
+	 *
+	 * @return string
+	 */
+	protected function get_shipping_datatype(): string {
+		return 'shipping';
+	}
+
+	/**
+	 * Get the settings datatype key
+	 *
+	 * @return string
+	 */
+	protected function get_settings_datatype(): string {
+		return 'settings';
+	}
+
+	/**
+	 * Get the default config for the sync mode.
+	 *
+	 * @return array[]
+	 */
+	protected function get_default_sync_mode(): array {
+		return [
+			$this->get_products_datatype() => [
+				'pull' => true,
+				'push' => true,
+			],
+			$this->get_coupons_datatype()  => [
+				'pull' => true,
+				'push' => true,
+			],
+			$this->get_shipping_datatype() => [
+				'pull' => true,
+				'push' => true,
+			],
+			$this->get_settings_datatype() => [
+				'pull' => true,
+				'push' => true,
+			],
+		];
```

Comment:
Now that we don't have the array indexes hardcoded, should we make this a little less repetitive by adding a default mode?

```suggestion
		$default_mode = [
			'pull' => true,
			'push' => true,
		];

		return [
			$this->get_products_datatype() => $default_mode,
			$this->get_coupons_datatype()  => $default_mode,
			$this->get_shipping_datatype() => $default_mode,
			$this->get_settings_datatype() => $default_mode,
		];
```

Reply:
Good suggestion. Added here https://github.com/woocommerce/google-listings-and-ads/pull/2804/commits/9ea4e80d7fb9543d4e0fa4945efc2477804a1792